src/MySqlConnector/ColumnReaders/BinaryBooleanColumnReader.cs

```diff
@@ -0,0 +1,20 @@
+namespace MySqlConnector.ColumnReaders;
+using System.Buffers.Text;
+using MySqlConnector.Protocol.Payloads;
+using MySqlConnector.Protocol.Serialization;
+using MySqlConnector.Utilities;
+
+internal sealed class BinaryBooleanColumnReader : IColumnReader
+{
+	internal static BinaryBooleanColumnReader Instance { get; } = new BinaryBooleanColumnReader();
```

Comment:
Singleton implementations of these classes is a good idea to reduce allocations 👍

Reply:
These `Instance` properties (or constructors for some types) are still logically part of the `public` API and should be `public` (even on an `internal` type).

Reply:
We can also use target-typed `new` here.

Will fix in https://github.com/mysql-net/MySqlConnector/pull/1332/commits/4d0560412933a87d351478b08052510743d3ae46.